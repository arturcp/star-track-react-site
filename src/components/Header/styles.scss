/* TODO: these variables are duplicated. See how to centralize them */
$main-color: #fff;
$secondary-color: #F8CD1B;

#top-header {
  position: fixed;
  display: inline-flex;
  top: 0;
  padding: 35px;
  background-color: $main-color;
  width: 100%;
  margin-bottom: 2px;
  box-shadow: #000 0 1px 3px 0px;

  .logo {
    width: 100px;
  }

  .description {
    margin-left: 80px;
  }

  h2 {
    margin-bottom: 0;
    font-weight: bold;
    font-size: 34px;
    color: $secondary-color;
    text-shadow: 0 0 #000;
  }
}
