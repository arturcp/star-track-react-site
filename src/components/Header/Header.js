import React from 'react';

import { Link } from 'react-router-dom';
import rankingIcon from '../../images/ranking.webp';
import logo from '../../images/rocket.png';
import './styles.scss';

const header = () => (
  <header id="top-header">
    <Link to="/">
      <img className="logo" src={logo} alt="Launching rocket to space" />
    </Link>
    <div className="description">
      <Link to="/">
        <h2>Star Track</h2>
        <br />
      </Link>
      <span>Use your knowledge to help Dr. Ken</span>
    </div>
    <div className="ranking-container">
      <Link to="/ranking">
        <img src={rankingIcon} alt="Ranking" />
      </Link>
    </div>
  </header>
);

export default header;
