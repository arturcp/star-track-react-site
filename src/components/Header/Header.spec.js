import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';

describe('Header UI', () => {
  it('should match the snapshot', () => {
    const component = renderer.create(
      <Router>
        <Header />
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
