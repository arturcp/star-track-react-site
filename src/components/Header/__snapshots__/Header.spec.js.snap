// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI should match the snapshot 1`] = `
<header
  id="top-header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Launching rocket to space"
      className="logo"
      src="rocket.png"
    />
  </a>
  <div
    className="description"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h2>
        Star Track
      </h2>
      <br />
    </a>
    <span>
      Use your knowledge to help Dr. Ken
    </span>
  </div>
  <div
    className="ranking-container"
  >
    <a
      href="/ranking"
      onClick={[Function]}
    >
      <img
        alt="Ranking"
        src="ranking.webp"
      />
    </a>
  </div>
</header>
`;
